models/bookingModel.ts

```diff
@@ -0,0 +1,90 @@
+import { PoolClient } from "pg";
+import pool from "../config/db";
+
+interface Booking {
+  booking_class: string;
+  user_id: number;
+  user_name: string;
+  email: string;
+  pnr: string;
+  train_id: number;
+  train_name: string;
+  journey_date: string;
+  booking_status: string;
+  booking_date: string;
+  total_amount: number;
+  refund_amount: number;
+  seat_number: string;
+}
+
+const getBookingByPNR = async (pnr: string): Promise<Booking | null> => {
+  const query = "SELECT * FROM booking WHERE pnr = $1";
+  const result = await pool.query(query, [pnr]);
+  return result.rows[0] || null;  
+};
+
+const cancelBooking = async (
+  pnr: string,
+  refundAmount: number,
+  seatNumbers: string
+): Promise<Booking> => {
+  const query1 = `

```

Comment:
Instead of using generic variable names like query1 or query2, use more descriptive names that provide a brief idea of the requirement. This will make the code more readable and easier to understand